Triggers.mediawiki

```diff
@@ -0,0 +1,78 @@
+<pre>
+  NEP: <to be assigned>
+  Title: Triggers for NeoContract
+  Author: Erik Zhang <erik@neo.org>
+  Type: Standard
+  Status: Draft
+  Created: 2017-10-16
+</pre>
+
+==Abstract==
+
+Trigger is a mechanism for triggering execution for smart contracts. This NEP defines 4 kinds of triggers, they are <code>Verification</code>, <code>VerificationR</code>, <code>Application</code> and <code>ApplicationR</code>.
+
+==Motivation==
+
+A blockchain that provides smart contract system should provide multiple triggers for the smart contracts running on it, makes them to function in different contexts.
+
+==Rationale==
+
+Currently, we have two kinds of triggers in NeoContract: <code>Verification</code> and <code>Application</code>. These two triggers make smart contracts to be able to verify the transactions and modify the states of the blockchain.
+
+But there is no way for smart contracts to refuse a transfer, or to modify the states of the blockchain while accepting a transfer. We need two new triggers to do it: <code>VerificationR</code> and <code>ApplicationR</code>.
+
+==Specification==
+
+We define 4 kinds of triggers: <code>Verification</code>, <code>VerificationR</code>, <code>Application</code> and <code>ApplicationR</code>.
+
+===Verification===
+
+This trigger indicates that the contract is being invoked as a verification function. The verification function can accept multiple parameters, and should return a boolean value that indicates the validity of the transaction or block.
```

Comment:
"This trigger" could be made clearer by changing it into "The <code>VerificationR</code> trigger indicates"